cmd/coder/urls.go

```diff
@@ -14,36 +16,163 @@ import (
 )
 
 type urlsCmd struct{}
+type createSubCmd struct {
+	access string
+}
+type delSubCmd struct{}
 
+// DevURL is the parsed json response record for a devURL from cemanager
 type DevURL struct {
+	ID     string `json:"id"`
 	URL    string `json:"url"`
 	Port   string `json:"port"`
 	Access string `json:"access"`
 }
 
-func (cmd urlsCmd) Spec() cli.CommandSpec {
+var urlAccessLevel = map[string]string{
+	//Remote API endpoint requires these in uppercase
+	"PRIVATE": "Only you can access",
+	"ORG":     "All members of your organization can access",
+	"AUTHED":  "Authenticated users can access",
+	"PUBLIC":  "Anyone on the internet can access this link",
+}
+
+func portIsValid(port string) bool {
+	p, err := strconv.ParseUint(port, 10, 16)
+	if p < 1 {
+		// port 0 means 'any free port', which we don't support
+		err = strconv.ErrRange
+	}
+	if err != nil {
+		fmt.Println("bad port")
+	}
+	return err == nil
+}
+
+func accessLevelIsValid(level string) bool {
+	_, ok := urlAccessLevel[level]
+	if !ok {
+		fmt.Println("Invalid access level")
+	}
+	return ok
+}
+
+func (sub *createSubCmd) RegisterFlags(fl *pflag.FlagSet) {
+	fl.StringVar(&sub.access, "a", "private", "[private | org | authed | public] set devurl access")
+}
+
+func (sub createSubCmd) Spec() cli.CommandSpec {
 	return cli.CommandSpec{
-		Name:  "urls",
-		Usage: "<env name>",
-		Desc:  "get all development urls for external access",
+		Name:  "create",
+		Usage: "<env name> <port> [--a access]",
+		Desc:  "create/update a devurl for external access",
 	}
```

Comment:
I made it just --a, do you want it fully-spelled out ie., --access ?

Reply:
I'd prefer `--acess` for the flag. If you also want the `-a` shorthand, we could use [fl.StringVarP](https://godoc.org/github.com/spf13/pflag#StringVarP)